tests/core.py

```diff
@@ -1795,6 +1796,12 @@ def test_variables(self):
         os.remove('variables2.json')
         os.remove('variables3.json')
 
+    def test_cli_version(self):
```

Comment:
I'm afraid this test is in the wrong place. tests/cli/test_cli.py is a better file.

I am also afraid of sys.stdout mocking. Python has a special mechanism for this. 
https://github.com/sekikn/incubator-airflow/blob/9ff0c4f79a009618215a2af60418f664168456f4/tests/cli/test_cli.py#L344-L352

Reply:
Thanks for the comment @mik-laj!

> I'm afraid this test is in the wrong place. tests/cli/test_cli.py is a better file.

The reason that I added the new test to test/core.py was it has many cli-related tests in its TestCli class. I have no objection to move the new test into tests/cli/test_cli.py, but is there any criteria to put a new cli-related test into it (or should we basically use tests/cli/test_cli.py in the future)?

> I am also afraid of sys.stdout mocking. Python has a special mechanism for this.

I didn't use `contextlib.redirect_stdout` because of concern for the compatibility with Python2, but I've just noticed [we're going to drop it on the master branch](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-3+Drop+support+for+Python+2). I'll address it on another PR, thanks :)